### PR TITLE
Docstrings refactoring

### PR DESCRIPTION
**Pull request by andrey-churkin**

### CODE REVIEW

**Comment by andrey-churkin** [comment]
https://github.com/openvinotoolkit/nncf/blob/983f6e16796a0cb68bb76d695e89ed5b39ad5cfc/nncf/common/pruning/utils.py#L109
@gadylshintr please provide a description.

**Comment by andrey-churkin** [comment]
https://github.com/openvinotoolkit/nncf/blob/983f6e16796a0cb68bb76d695e89ed5b39ad5cfc/nncf/common/pruning/pruning_node_selector.py#L189
@AlexanderDokuchaev please provide a description.

**Comment by andrey-churkin** [comment]
https://github.com/openvinotoolkit/nncf/blob/983f6e16796a0cb68bb76d695e89ed5b39ad5cfc/nncf/common/pruning/model_analysis.py#L173
@mkaglins please provide a description.

File: nncf/common/pruning/utils.py [hunk 1]
```diff
@@ -100,10 +103,11 @@ def get_rounded_pruned_element_number(total: int, sparsity_rate: float, multiple
     Calculates number of sparsified elements (approximately sparsity rate) from total such as
     number of remaining items will be multiple of some value.
     Always rounds number of remaining elements up.
-    :param total: total elements number
-    :param sparsity_rate: prorortion of zero elements in total.
+
+    :param total: Total elements number.
+    :param sparsity_rate: Prorortion of zero elements in total.
     :param multiple_of:
```
Comment (gadylshintr) [inline nncf/common/pruning/utils.py:109]:
```suggestion
    :param multiple_of: number of remaining elements must be a multiple of `multiple_of`
```

**Review by alexsu52 — COMMENTED** [review]
@andrey-churkin, good job!

**Comment by mkaglins** [comment]
> https://github.com/openvinotoolkit/nncf/blob/983f6e16796a0cb68bb76d695e89ed5b39ad5cfc/nncf/common/pruning/model_analysis.py#L173
> 
> 
> @mkaglins please provide a description.

`:param clusterization: Clusterization of nodes to work with. `

And typo in **merged** above: https://github.com/openvinotoolkit/nncf/blob/983f6e16796a0cb68bb76d695e89ed5b39ad5cfc/nncf/common/pruning/model_analysis.py#L172

**Review by vshampor — APPROVED** [review]

File: nncf/common/graph/graph.py [hunk 2]
```diff
@@ -61,8 +62,9 @@ def __eq__(self, other):
 class NNCFGraph:
     """
     Wrapper over a regular directed acyclic graph that represents a control flow/execution graph of a DNN
-    providing some useful methods for graph traverse
+    providing some useful methods for graph traverse.
```
Comment (vanyalzr) [inline nncf/common/graph/graph.py:65]: traverse -> traversal

File: nncf/common/graph/graph.py [hunk 3]
```diff
@@ -170,10 +176,11 @@ def traverse_graph(self,
...
         """
-        Traverses graph up or down starting form `curr_node` node
-        :param curr_node: node from which traverse is started
-        :param traverse_function: function describing condition of traverse continuation/termination
-        :param traverse_forward: flag specifying direction of traverse
+        Traverses graph up or down starting form `curr_node` node.
+
+        :param curr_node: Node from which traverse is started.
+        :param traverse_function: Function describing condition of traverse continuation/termination.
+        :param traverse_forward: Flag specifying direction of traverse.
```
Comment (vanyalzr) [inline nncf/common/graph/graph.py:183]: traversal in each of the three lines above

File: nncf/common/pruning/model_analysis.py [hunk 4]
```diff
@@ -219,18 +228,19 @@ def cluster_special_ops(graph: NNCFGraph, special_types: List[str], identity_typ
...
-    1. Set attribute accept_pruned_input to all nodes. This attribute shows can this node potentially get
-     pruned input or node.
-    2.  Calculate can_prune attribute for all nodes by propagating accept_pruned_input up
-     (from the result of the network to the inputs). Node can be pruned if all outputs of this node accept
-      pruned input and all outputs can be pruned.
-    3. Propagates can_prune down from input nodes to the outputs.
-
-    As a result, all nodes marked by the can_prune attribute as potentially prunable or not.
+        1. Set attribute `accept_pruned_input` to all nodes. This attribute shows can this node potentially get
+        pruned input or node.
```
Comment (vanyalzr) [inline nncf/common/pruning/model_analysis.py:235]: ... shows whether this node can potentially get pruned input

File: nncf/common/pruning/model_analysis.py [hunk 5]
```diff
@@ -219,18 +228,19 @@ def cluster_special_ops(graph: NNCFGraph, special_types: List[str], identity_typ
...
+        2.  Calculate `can_prune` attribute for all nodes by propagating `accept_pruned_input` up
+        (from the result of the network to the inputs). Node can be pruned if all outputs of this node accept
+        pruned input and all outputs can be pruned.
+        3. Propagates `can_prune` down from input nodes to the outputs.
+
+    As a result, all nodes marked by the `can_prune` attribute as potentially prunable or not.
```
Comment (vanyalzr) [inline nncf/common/pruning/model_analysis.py:241]: all nodes are marked

File: nncf/common/pruning/model_analysis.py [hunk 6]
```diff
@@ -244,29 +254,34 @@ def __init__(self, graph: NNCFGraph,
...
+        self.can_prune = {idx: True for idx in self.graph.get_all_node_ids()}
+        self.accept_pruned_input = {idx: True for idx in self.graph.get_all_node_ids()}
 
     def node_propagate_can_prune_attr(self, nncf_node: NNCFNode) -> bool:
         """
-        Whether node propagates can_prune attr through. That means a node can propagate pruning mask
-         (for example,  activations propagate mask, but convolutions stop mask propagation)
-        :param nncf_node: node to work with
-        :return: bool: propagates this node can_prune throw or not
+        Whether node propagates `can_prune` attr through. That means a node can propagate pruning mask
```
Comment (vanyalzr) [inline nncf/common/pruning/model_analysis.py:262]: Whether the node can propagate the `can_prune` attr

File: nncf/common/pruning/pruning_node_selector.py [hunk 7]
```diff
@@ -91,15 +92,15 @@ def __init__(self,
...
+            3. Add remaining single nodes
+            4. Unite clusters for Conv + Depthwise conv (should be pruned together too)
+            5. Checks for groups (all nodes in group can prune or all group can't be pruned)
         Return groups of modules that should be pruned together.
-        :param graph: graph to work with
-        and their initialization parameters as values
-        :return: clusterisation of pruned nodes
+
+        :param graph: Fraph to work with and their initialization parameters as values.
+        :return: Clusterisation of pruned nodes.
```
Comment (vanyalzr) [inline nncf/common/pruning/pruning_node_selector.py:103]: Clusterization (this spelling is used in other places)

File: nncf/common/graph/graph.py (hunk 2 above)
Comment (andrey-churkin) [reply nncf/common/graph/graph.py]: Thanks! I fixed it.

File: nncf/common/graph/graph.py (hunk 3 above)
Comment (andrey-churkin) [reply nncf/common/graph/graph.py]: Thanks! I fixed it.

File: nncf/common/pruning/model_analysis.py (hunk 4 above)
Comment (andrey-churkin) [reply nncf/common/pruning/model_analysis.py]: Thanks! I fixed it.

File: nncf/common/pruning/model_analysis.py (hunk 5 above)
Comment (andrey-churkin) [reply nncf/common/pruning/model_analysis.py]: Thanks! I fixed it.

File: nncf/common/pruning/model_analysis.py (hunk 6 above)
Comment (andrey-churkin) [reply nncf/common/pruning/model_analysis.py]: Thanks! I fixed it.

File: nncf/common/pruning/pruning_node_selector.py (hunk 7 above)
Comment (andrey-churkin) [reply nncf/common/pruning/pruning_node_selector.py]: Thanks! I fixed it.

**Review by alexsu52 — APPROVED** [review]

**Pull request merged**